prestashop1.7/controllers/admin/AdminTawktoController.php

```diff
@@ -239,40 +279,97 @@ public function ajaxProcessSetVisibility()
         ];
 
-        $options = Tools::getValue('options');
-        if (!empty($options)) {
-            $options = explode('&', $options);
-            foreach ($options as $post) {
-                [$column, $value] = explode('=', $post);
-                switch ($column) {
-                    case 'hide_oncustom':
-                    case 'show_oncustom':
-                        // replace newlines and returns with comma, and convert to array for
-                        // saving
-                        $value = urldecode($value);
-                        $value = str_ireplace(["\r\n", "\r", "\n"], ',', $value);
-                        if (!empty($value)) {
-                            $value = explode(',', $value);
-                            $jsonOpts[$column] = json_encode($value);
-                        }
+        if (empty($params)) {
+            return $jsonOpts;
+        }
+
+        parse_str($params, $options);
+        foreach ($options as $column => $value) {
+            switch ($column) {
+                case 'hide_oncustom':
+                case 'show_oncustom':
+                    // replace newlines and returns with comma, and convert to array for saving
+                    $value = urldecode($value);
+                    $value = str_ireplace(["\r\n", "\r", "\n"], ',', $value);
+                    if (!empty($value)) {
+                        $value = explode(',', $value);
+                        $jsonOpts[$column] = json_encode($value);
+                    }
+                    break;
+
+                case 'show_onfrontpage':
+                case 'show_oncategory':
+                case 'show_onproduct':
+                case 'always_display':
+                case 'enable_visitor_recognition':
+                    $jsonOpts[$column] = ($value == 1);
+                    break;
+
+                case 'js_api_key':
+                    if ($value === self::NO_CHANGE) {
+                        unset($jsonOpts['js_api_key']);
                         break;
+                    }
 
-                    case 'show_onfrontpage':
-                    case 'show_oncategory':
-                    case 'show_onproduct':
-                    case 'always_display':
-                    case 'enable_visitor_recognition':
-                        $jsonOpts[$column] = ($value == 1);
+                    if ($value === '') {
                         break;
-                }
+                    }
+
+                    try {
+                        if (strlen(trim($value)) !== 40) {
+                            throw new Exception('Invalid API key. Please provide value with 40 characters');
+                        }
+
+                        $jsonOpts['js_api_key'] = $this->encryptData($value);
+                    } catch (Exception $e) {
+                        unset($jsonOpts['js_api_key']);
+
+                        throw new Exception('Javascript API Key: ' . $e->getMessage());
```

Comment:
We may be bleeding internals here by using `getMessage` here. Is there a way we can log this and return generic error>